homeassistant/components/squeezebox/__init__.py

```diff
@@ -1 +1,86 @@
-"""The squeezebox component."""
+"""The Logitech Squeezebox integration."""
+
+import asyncio
+import logging
+
+from pysqueezebox import async_discover
+
+from homeassistant.components.media_player import DOMAIN as MP_DOMAIN
+from homeassistant.config_entries import SOURCE_DISCOVERY, ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, EVENT_HOMEASSISTANT_START
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, ENTRY_PLAYERS, KNOWN_PLAYERS, PLAYER_DISCOVERY_UNSUB
+
+_LOGGER = logging.getLogger(__name__)
+
+DISCOVERY_TASK = "discovery_task"
+
+
+async def start_server_discovery(hass):
+    """Start a server discovery task."""
+
+    def _discovered_server(server):
+        asyncio.create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_DISCOVERY},
+                data={
+                    CONF_HOST: server.host,
+                    CONF_PORT: int(server.port),
+                    "uuid": server.uuid,
+                },
+            )
+        )
+
+    hass.data.setdefault(DOMAIN, {})
+    if DISCOVERY_TASK not in hass.data[DOMAIN]:
+        _LOGGER.debug("Adding server discovery task for squeezebox")
+        hass.data[DOMAIN][DISCOVERY_TASK] = hass.async_create_task(
+            async_discover(_discovered_server)
+        )
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Logitech Squeezebox component."""
+    if hass.is_running:
+        asyncio.create_task(start_server_discovery(hass))
```

Comment:
So that instead of loading the `discovery:` integration, you can load `squeezebox:` in `configuration.yaml` and auto-discover servers. Since `discovery` is deprecated and most other things can be discovered through `ssdp` and `zeroconf`, I thought this was useful.

I run it when setting up a config entry too, so that you automatically begin discovering squeezebox servers once you first load the integration.

`start_server_discovery` checks if server discovery is already running so we won't end up with multiple copies of this running.

Reply:
We're deprecating the platform section in configuration.yaml here so we shouldn't add new features that depend on some other part of configuration.yaml.

Reply:
Ok, then I will run the internal discovery for servers only if there is a current config entry. This and other suggested changes in PR #37030